pkg/distributor/validator.go

```diff
@@ -13,6 +13,10 @@ import (
 	"github.com/grafana/loki/pkg/validation"
 )
 
+const (
+	timeFormat = time.RFC3339
```

Comment:
How do we ensure that this is the format that will be used when passing `time.Time` values to the `fmt.Sprintf` function? I believe it comes down to locale, no?
This feels a bit too loose for me.

Reply:
Hmm. no we cannot control how printf formats it (unless we have to create custom type with `String()` function i guess). I understand what you mean, the timestamps can be sent to printf without calling explicitly `.Format()` causing inconsistency. 
I think I can remove it.

Only problem is, the extra `monotonic` data is interrupted in the time string when using `Printf()`.

Usually when you pass the time object (`time.Now()` for example) directly to `Printf` with `%s` as format string, then it prints it with `m=`  e.g:
```
2022-01-05 09:03:08.757393017 +0100 CET m=-17999.987715510
```

The problem is we loose that information when we convert it to Unix timestamp e.g: `t.UnixNano()` (it's just int64)

Even when you convert back from UnixNano -> time.Time via `time.UnixNano()`, that information is lost (which is what happening in our [validator function.](https://github.com/grafana/loki/blob/main/pkg/distributor/validator.go#L47-L48))

So in the assertion, time A (used in test) will have `m=` information, where as time B (used in actual validator function, that is converted from unixnano (int64), won't have that information)

NOTE; `monotonic` information is only for debugging purposes, and the time only in the tests is having this information. Normally, checking the actual response from HTTP endpoint, don't have this information. e.g:
```
2022-01-05 19:03:08.757393017 +0100 CET 
```

There are two options I see, 
1. Retain the actual time object rather than converting it into int64 in `validationContext` 
2. Or format time that is sent in the response in consistent way (this is what I tried here).

(1) seems to be big of a change for this PR.  I see all the types inside that struct is primary types. Not sure it's worth changing that?

Reply:
I think we should go for option 2

Reply:
which is basically what I did here.
https://github.com/grafana/loki/blob/41293185b11bd5033896eec3d423a4821c929f51/pkg/distributor/validator.go#L64-L66

Reply:
Ah, I didn't notice that your tests and the implementation were both using the same `timeFormat` variable - apologies; so nothing should be using the locale AFAICS then?

Reply:
nope. no locale involved.